client/src/streams/stream.rs

```diff
@@ -1,122 +1,248 @@
         };
 
-        let number_of_packets = self.rtp_packets.len() as f64;
-        let last_sequence_number = rtp.sequence_number as f64;
-        let expected_number_of_packets = last_sequence_number - first_sequence_number as f64 + 1.0;
-        self.lost_percentage = 100.0 - (number_of_packets / expected_number_of_packets) * 100.0;
+        // update_prev_lost first_X/last_X updated
+        self.update_prev_lost(&mut rtp_info);
+        self.rtp_packets.push(rtp_info);
     }
 
-    fn calculate_jitter(&mut self, packet: &Packet) {
-        let SessionPacket::Rtp(ref rtp) = packet.contents else {
-            unreachable!();
+    fn update_jitter(&self, rtp_info: &mut RtpInfo) {
+        let Some(clock_rate) = rtp_info.packet.payload_type.clock_rate else {
+            return;
         };
 
-        let Some(last_timestamp) = self.previous_timestamp else {
-            self.previous_timestamp = Some(packet.timestamp);
-            self.previous_rtp_timestamp = Some(rtp.timestamp as f64);
+        let Some(prev_rtp_info) = self.rtp_packets.last() else {
+            // rtp_info is the first packet
+            rtp_info.jitter = Some(0.0);
             return;
         };
 
-        if rtp.payload_type.clock_rate.is_none() || rtp.payload_type.id != self.payload_type {
-            self.payload_type = rtp.payload_type.id;
-            self.jitter_in_ms = 0.0;
-            self.jitter_history.clear()
-        } else {
-            // RFC 3550
-            let clock_rate = rtp.payload_type.clock_rate.unwrap();
-            let unit_timestamp = 1.0 / clock_rate as f64;
-            let arrival_time_difference_result = packet.timestamp.checked_sub(last_timestamp);
-            if let Some(arrival_time_difference) = arrival_time_difference_result {
-                let timestamp_difference = rtp.timestamp as f64 * unit_timestamp
-                    - self.previous_rtp_timestamp.unwrap() * unit_timestamp;
-                let d_in_sec = arrival_time_difference.as_secs_f64() - timestamp_difference;
-                let d_in_ms = d_in_sec.abs() * 1000.0;
-
-                self.jitter_in_ms = self.jitter_in_ms + (d_in_ms - self.jitter_in_ms) / 16.0;
-                self.jitter_history.push(self.jitter_in_ms);
-            }
+        let is_new = rtp_info.packet.payload_type.id != prev_rtp_info.packet.payload_type.id;
+        if is_new {
+            rtp_info.jitter = Some(0.0);
+            return;
         }
 
-        self.previous_timestamp = Some(packet.timestamp);
-        self.previous_rtp_timestamp = Some(rtp.timestamp as f64);
+        let unit = 1.0 / clock_rate as f64;
+        let arrival_diff = rtp_info
+            .time
+            .checked_sub(prev_rtp_info.time)
+            .unwrap()
+            .as_secs_f64();
+        let rtp_timestamp_diff =
+            (rtp_info.packet.timestamp as i64 - prev_rtp_info.packet.timestamp as i64) as f64;
+        let diff = arrival_diff - rtp_timestamp_diff * unit;
```

Comment:
```suggestion
        let diff = (arrival_diff - rtp_timestamp_diff).abs() * unit;
```

Reply:
I don't think this should be an absolute value, but I'm going to take a closer look at the jitter when refactoring the Streams table view.